lib/govuk_app_config/govuk_error/configuration.rb

```diff
@@ -51,5 +52,19 @@ def ignore_excluded_exceptions_in_data_sync
         !(data_sync.in_progress? && data_sync_ignored_error)
       }
     end
+
+    def increment_govuk_statsd_counters
+      lambda { |error_or_event, _hint|
+        GovukStatsd.increment("errors_occurred")
+        GovukStatsd.increment("error_types.#{error_or_event.class.name.demodulize.underscore}")
```

Comment:
Argh - sorry to throw another spanner in the works, but I've just realised we wouldn't want to increment these values unless we actually end up logging the error. As it stands, we could set our own `before_send=` lambda to check an error type and decide to return `nil`, but by that point we've already evaluated `increment_govuk_statsd_counters` and thus would have incorrectly incremented our global error count.

This can be in its own separate commit: could you write a test that ensures that `increment_govuk_statsd_counters` is the last callback that gets called? I'm _hoping_ it's actually not too much work - should just be a case of reversing 

`default_before_send_actions.call(error_or_event, hint) && closure.call(error_or_event, hint)` 
=>
`closure.call(error_or_event, hint) && default_before_send_actions.call(error_or_event, hint)`

Reply:
Hm, I think this needs some caution. If the user-defined callback returns nil then the default callback won't be run — but returning nil seems fairly common and innocuous (e.g., `GovukStatsd.increment('…')` returns nil which is how I realised).

The documentation for `before_send` says that it ought to return the input if all is well or nil if not, so it's something that _should_ be accounted for when writing the callback, but it also seems like it would be easy not to notice that the counters were no longer being incremented. Perhaps I'm overthinking it though.

Reply:
Yes, we may want to make it explicit in documentation somewhere, though there's only one instance of `before_send` being used on GOV.UK and they've [explicitly returned the error](https://github.com/alphagov/govuk-coronavirus-vulnerable-people-form/blob/37fd197b2d94e55f3ed759b381181d282c4a18c8/config/application.rb#L33), so I'm hoping this is a non-problem :)